### PR TITLE
Make getPlayerCount async

### PR DESCRIPTION
**Pull request by kberg**
This required a couple of external changes. For instance, introducing an
Async version of a handler, so tests could wait for results.

Also fixed an irritating issue, looks like I did a poor job of
saving and restoring the fake database test. Ha.

### CODE REVIEW

File: src/routes/Handler.ts [hunk 1]
```diff
@@ -44,13 +44,28 @@ export abstract class Handler implements IHandler {
     }
   }
 
-  get(req: http.IncomingMessage, res: http.ServerResponse, ctx: IContext): void {
+  get(req: http.IncomingMessage, res: http.ServerResponse, ctx: IContext): void | Promise<void> {
```
Comment (bafolts) [inline src/routes/Handler.ts:47]: With the `void | Promise<void>` syntax one can use `await handler.get()`. Either will work with `void | Promise<void>`. This also avoids having to return `Promise.resolve` in source. Was this change needed?

Comment (kberg) [reply src/routes/Handler.ts:47]: This is a completely fair question.

My goal is to move all the handlers to Promise<void> but this was the only way I could think of doing it in transit. What do you think?

Comment (bafolts) [reply src/routes/Handler.ts:47]: Using `void | Promise<void>` should be fine. When we are using a `Promise` I would use `await` versus `.then`. If using `await` there is more flexibility as you can `await` something that isn't a `Promise` and a `Promise` will be created. If a route doesn't do anything asynchronous it could then not have to have `return Promise.resolve()` at the end. This is less code we have to write and maintain.

I don't think this is a huge deal but we will end up with a lot of functions having `return Promise.resolve()` when they could still be `void`. The caller should be using `await` so it doesn't matter if we return void.

**Review by bafolts — APPROVED** [review]

File: src/routes/Handler.ts (hunk 1 above)
Comment (kberg) [reply src/routes/Handler.ts:47]: I think I'm still missing how these things all work together in a fundamental mechanical way. I'm still for the most part pattern matching and hoping the tests work. Do you have a recommendation for something that can help me really internalize all of this?
